src/main/rst/examples/Chapter4/Java/ProxySetupFirefoxMarionette.java

```diff
@@ -0,0 +1,15 @@
+String PROXY = "localhost";
+int PORT = 8080;
+
+com.google.gson.JsonObject json = new com.google.gson.JsonObject();
+json.addProperty("proxyType", "MANUAL");
+json.addProperty("httpProxy", PROXY);
+json.addProperty("httpProxyPort", PORT);
+json.addProperty("sslProxy", PROXY);
+json.addProperty("sslProxyPort", PORT);
+
+DesiredCapabilities desired = DesiredCapabilities.firefox();
+DesiredCapabilities required = new DesiredCapabilities();
+required.setCapability("proxy", json);
+
+driver = new FirefoxDriver(desired, required);
```

Comment:
Hi @lukeis,

You are right about this, but there is more. This example does not work with the constructor new FirefoxDriver(desired, required). It passes the capabilities through FirefoxDriver.extractProfile, that is calling populateProfile, that is adding "firefoxOptions" to capabilities.

> desiredCapabilities=Capabilities [{marionette=true, firefoxOptions=org.openqa.selenium.firefox.FirefoxOptions@10dba097, browserName=firefox, version=, platform=ANY}]...

The response from the other side is 

> org.openqa.selenium.SessionNotCreatedException: Not all requiredCapabilities could be met: ["[object Object] does not equal undefined"]

It works with the service constructor 

> FirefoxDriver(GeckoDriverService driverService, Capabilities desiredCapabilities, Capabilities requiredCapabilities)

This does not change the capabilities, and it works.

So I will change the example to 

`		GeckoDriverService service =new GeckoDriverService.Builder()
				  .usingDriverExecutable(new File("path to geckodriver"))
				  .usingAnyFreePort()
				  .usingAnyFreePort()
				  .build();
			  service.start();   
		WebDriver driver = new FirefoxDriver(service, cap, cap);`